lib/jellyfish/notifier.ex

```diff
@@ -112,4 +169,73 @@ defmodule Jellyfish.Notifier do
         error
     end
   end
+
+  defp handle_notification(%Authenticated{}, state) do
+    send(state.caller_pid, {:jellyfish, :authenticated})
+    state
+  end
+
+  defp handle_notification(%RoomNotFound{id: id}, state) do
+    {pid, state} = pop_in(state.pending_subscriptions[id])
+
+    send(pid, {:jellyfish, {:subscribe_answer, {:error, :room_not_found}}})
+    state
+  end
+
+  defp handle_notification(%mod{} = room, state) when mod in [RoomState, RoomsState] do
+    {room_id, room} =
+      case mod do
+        RoomState -> {room.id, from_room_state_message(room)}
+        RoomsState -> {:all, Enum.map(room.rooms, &from_room_state_message/1)}
+      end
+
+    {pid, state} = pop_in(state.pending_subscriptions[room_id])
+
+    Process.monitor(pid)
+
+    state =
+      update_in(state.subscriptions[room_id], fn
+        nil -> MapSet.new([pid])
+        set -> MapSet.put(set, pid)
+      end)
+
+    send(pid, {:jellyfish, {:subscribe_answer, {:ok, room}}})
+    state
+  end
+
+  defp handle_notification(%{room_id: room_id} = message, state) do
+    state.subscriptions
+    |> Map.take([:all, room_id])
+    |> Map.values()
+    |> Enum.reduce(fn pids, acc -> MapSet.union(pids, acc) end)
+    |> Enum.each(&send(&1, {:jellyfish, Notification.to_notification(message)}))
+
+    state
+  end
+
+  defp from_room_state_message(msg) do
+    peers =
+      msg.peers
+      |> Enum.map(
+        &%Peer{id: &1.id, type: from_proto_type(&1.type), status: from_proto_status(&1.status)}
+      )
+
+    components =
+      msg.components
+      |> Enum.map(&%Component{id: &1.id, type: from_proto_type(&1.type)})
```

Comment:
IMO these should be implemented in Peer and Component, i.e.

```suggestion
      |> Enum.map(&Peer.from_proto/1)

    components =
      msg.components
      |> Enum.map(&Component.from_proto/1)
```